### PR TITLE
[TECH] JWT parser

### PR DESCRIPTION
**Pull request by tensenrey**

### CODE REVIEW

File: src/jwt-parser/jwt-parser.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    save(){
+        localStorage.setItem(this.jwtKey, JSON.stringify(this.data));
+    }
+    get(){
+        return JSON.parse(localStorage.getItem(this.jwtKey));
+    }
+}
+
+export default new JWT();
```
Comment (mtalstykh-exadel) [inline src/jwt-parser/jwt-parser.js]: move it to the src/utils/jwt-parser.js

File: src/components/counter/counter.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
         <button onClick = {()=>{dispatch(increment());}}>INCREMENT BY 1</button>
-      </div>
-      <div>
         <button onClick = {()=>{dispatch(decrement());}}>DECREMENT BY 1</button>
-      </div>
         <button onClick = {()=>{dispatch(reset());}}>RESET</button>
+        <button onClick = {()=>{JWT.parse("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE2MjYxNjA3OTQsImV4cCI6MTY1NzY5Njc5NCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSIsInJvbGUiOiJ1c2VyIiwiaW1nIjoiaHR0cHM6Ly9taXJwb3ppdGl2YS5ydS93cC1jb250ZW50L3VwbG9hZHMvMjAxOS8xMS8xNDcyMDQyNzE5XzE1LmpwZyIsIm5hbWUiOiJBbGJlcnQgRWluc3RlaW4ifQ.kJsZuRrqspFXaABSO3jOVzB4RKdMIpyhisp7CO2se8c");}}>
+          parse jwt
+        </button>
+        <button onClick = {()=>{console.log(JWT.get());}}> get JWT </button>
```
Comment (mtalstykh-exadel) [inline src/components/counter/counter.js]: resolve conflicts

**Review by mtalstykh-exadel — DISMISSED** [review]

File: src/App.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ function App() {
         <PrivateRoute path="/buttons"><Buttons /></PrivateRoute>
         <PrivateRoute path="/counter"><Counter /></PrivateRoute>
         <PrivateRoute path="/profile"><Profile /></PrivateRoute>
+        <PrivateRoute path="/jwt"><DemoJWT /></PrivateRoute>
```
Comment (mtalstykh-exadel) [inline src/App.js:20]: route pages should be moved to the src/pages/[pageName]/[pageName].jsx

File: src/components/profile/profile.jsx [hunk 4]
```diff
@@ -27,6 +27,9 @@ const Profile = () => {
 		<Button variant="contained" color="primary" disableElevation className={classes.testButtons}>
 			<Link to='/buttons'>Buttons</Link>
 		</Button>
+		<Button variant="contained" color="primary" disableElevation className={classes.testButtons}>
+			<Link to='/jwt'>JWT</Link>
+		</Button>
```
Comment (mtalstykh-exadel) [inline src/components/profile/profile.jsx:32]: same for the profile page

File: src/components/jwt/demo-jwt.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import JWT from '../../utils/jwt-parser.js';
+export default function DemoJWT(){
+    return(
+        <>
+            <button onClick={ ()=>JWT.parse("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE2MjYxNjk2OTksImV4cCI6MTY1NzcwNTY5OSwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSIsIm5hbWUiOiJKb2hubnkgUm9ja2V0IiwiZW1haWwiOiJkYXNkQG1haWwuZ2xvYmFsIiwicm9sZSI6InVzZXIifQ.HTDnBXJxkmky_YqOOw_L5nu2Xn6A-UZQp-5u2WBPZIg") }>
+                1) parse jwt
+            </button>
+            <button onClick={ ()=>console.log(JWT.get()) }>2) get jwt</button>
```
Comment (fshovchko) [inline src/components/jwt/demo-jwt.jsx]: remove console.log()

**Review by fshovchko — DISMISSED** [review]

**Review by fshovchko — DISMISSED** [review]

File: src/App.js [hunk 6]
```diff
@@ -5,21 +5,23 @@ import './App.scss';
 import Buttons from './components/buttons/Buttons';
 import Counter from './components/counter/counter';
 import PrivateRoute from './utils/privateRoute';
-import Profile from './components/profile/profile';
+import Profile from './pages/profile/profile';
+import DemoJWT from './pages/jwt/jwt.jsx';
+function App() {
```
Comment (mtalstykh-exadel) [inline src/App.js]: please, at leat one empty string)

File: src/pages/jwt/jwt.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        <>
+            <div> JWTDATA = <br/> {jwtData}</div>
+            <button onClick={ ()=>JWT.parse("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE2MjYxNjk2OTksImV4cCI6MTY1NzcwNTY5OSwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSIsIm5hbWUiOiJKb2hubnkgUm9ja2V0IiwiZW1haWwiOiJkYXNkQG1haWwuZ2xvYmFsIiwicm9sZSI6InVzZXIifQ.HTDnBXJxkmky_YqOOw_L5nu2Xn6A-UZQp-5u2WBPZIg") }>
+                1) parse jwt
+            </button>
+            <button onClick={ ()=> setJwtData( JSON.stringify(JWT.get()) ) }>2) get jwt</button> 
+        </>
+    );
+}
+
```
Comment (mtalstykh-exadel) [inline src/pages/jwt/jwt.jsx]: fix spaces. should be 2 not 4

File: src/pages/login/login.jsx [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import LoginForm from '../../components/login/loginForm/LoginForm';
-import logo from '../../assets/images/logo/logoText.png';
+import logo from '../../assets/images/logo/logoText.svg';
```
Comment (mtalstykh-exadel) [inline src/pages/login/login.jsx]: remove it and leave for an another ticket

File: src/utils/jwt-parser.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    get(){
+        return JSON.parse(localStorage.getItem(this.jwtKey));
+    }
+    remove(){
+        localStorage.removeItem(this.jwtKey);
+    }
+}
+
+export default new JWT();
```
Comment (mtalstykh-exadel) [inline src/utils/jwt-parser.js:23]: 2 spaces...

**Review by fshovchko — DISMISSED** [review]

**Review by fshovchko — APPROVED** [review]

**Review by mtalstykh-exadel — APPROVED** [review]

**Pull request merged**